mmv1/products/gkeonprem/BareMetalCluster.yaml

```diff
@@ -0,0 +1,943 @@
+          http://(username:password@)domain these will be rejected by the server.
+      - !ruby/object:Api::Type::Array
+        name: "noProxy"
+        item_type: Api::Type::String
+        description: |
+          A list of IPs, hostnames, and domains that should skip the proxy.
+          Examples: ["127.0.0.1", "example.com", ".corp", "localhost"].
+  - !ruby/object:Api::Type::NestedObject
+    name: "clusterOperations"
+    description: |
+      Specifies the User Cluster's observability infrastructure.
+    properties:
+      - !ruby/object:Api::Type::Boolean
+        name: "enableApplicationLogs"
+        description: |
+          Whether collection of application logs/metrics should be enabled (in addition to system logs/metrics).
+  - !ruby/object:Api::Type::NestedObject
+    name: "maintenanceConfig"
+    description: |
+      Specifies the workload node configurations.
+    properties:
+      - !ruby/object:Api::Type::Array
+        name: "maintenanceAddressCidrBlocks"
+        required: true
+        item_type: Api::Type::String
+        description: |
+          All IPv4 address from these ranges will be placed into maintenance mode.
+          Nodes in maintenance mode will be cordoned and drained. When both of these
+          are true, the "baremetal.cluster.gke.io/maintenance" annotation will be set
+          on the node resource.
+  - !ruby/object:Api::Type::NestedObject
+    name: "nodeConfig"
+    description: |
+      Specifies the workload node configurations.
+    properties:
+      - !ruby/object:Api::Type::Integer
+        name: "maxPodsPerNode"
+        description: |
+          The maximum number of pods a node can run. The size of the CIDR range
+          assigned to the node will be derived from this parameter.
+        default_from_api: true
+      - !ruby/object:Api::Type::Enum
+        name: "containerRuntime"
+        description: |
+          The available runtimes that can be used to run containers in a Bare Metal User Cluster.
+        values:
+          - CONTAINER_RUNTIME_UNSPECIFIED
+          - DOCKER
+          - CONTAINERD
+        default_from_api: true
+  - !ruby/object:Api::Type::NestedObject
+    name: "nodeAccessConfig"
+    description: |
+      Specifies the node access related settings for the bare metal user cluster.
+    properties:
+      - !ruby/object:Api::Type::String
+        name: "loginUser"
+        required: false
+        description: |
+          LoginUser is the user name used to access node machines.
+          It defaults to "root" if not set.
+        default_from_api: true
+  - !ruby/object:Api::Type::NestedObject
+    name: "osEnvironmentConfig"
+    description: |
+      OS environment related configurations.
+    properties:
+      - !ruby/object:Api::Type::Boolean
+        name: "packageRepoExcluded"
+        required: true
+        description: |
+          Whether the package repo should not be included when initializing
+          bare metal machines.
```

Comment:
This field doesn't seem to be documented externally
https://cloud.google.com/anthos/clusters/docs/on-prem-api/reference/rest/v1/projects.locations.bareMetalClusters

Reply:
The external document is out of date. Please refer to the internal one. [https://cloud.devsite.corp.google.com/anthos/clusters/docs/on-prem-api/reference/rest/v1/projects.locations.bareMetalClusters](https://cloud.devsite.corp.google.com/anthos/clusters/docs/on-prem-api/reference/rest/v1/projects.locations.bareMetalClusters#baremetalcontrolplaneconfig)